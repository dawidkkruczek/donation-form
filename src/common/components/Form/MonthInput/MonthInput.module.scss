.root {
  align-items: center;
  background: var(--color-white);
  border-radius: 4px;
  border: 1px solid var(--color-stroke);
  display: flex;
  justify-content: center;
  height: 62px;
  padding: 12px;
  position: relative;
  width: 100%;
}

.input {
  display: none;
}

.value {
  line-height: 1.25;
  text-align: center;
}

.valueMonth,
.valueYear {
  display: block;
}

.valueMonth {
  font-family: 'Rubik';
  font-weight: 500;
  font-size: 16px;
}

.valueYear {
  font-family: 'Work Sans';
  font-size: 12px;
}

.arrow {
  align-items: center;
  background-color: transparent;
  border-radius: 5px;
  border: 0;
  display: flex;
  height: 24px;
  justify-content: center;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  transition: all 0.5s ease;
  width: 24px;
  z-index: 2;

  &:not(:disabled) {
    cursor: pointer;

    &:hover {
      background-color: var(--color-gray-lighter);
    }

    &:active {
      background-color: var(--color-gray-light);
    }
  }
}

.arrowPrev {
  left: 12px;
}

.arrowNext {
  right: 12px;
  transform: translateY(-50%) rotate(180deg);
}

.arrowIcon {
  color: var(--color-midnight-gray);
}
