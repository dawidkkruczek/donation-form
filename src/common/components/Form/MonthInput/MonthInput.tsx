import React, { ChangeEvent, ReactElement } from 'react';
import classNames from 'classnames';
import Input, { Props as InputProps } from 'common/components/Form/Input/Input';
import moment from 'moment';
import { ReactComponent as ArrowComponent } from './assets/arrow.svg';
import styles from './MonthInput.module.scss';

export interface Props extends Omit<InputProps, 'onChange'> {
  defaultValue?: string;
  onChange?: (value: string) => void;
  value?: string;
}

function AmountInput({
  className,
  onChange,
  ...inputProps
}: Props): ReactElement {
  const { min, value } = inputProps;

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    onChange(e.target.value);
  };

  const handlePrevClick = () => {
    onChange(moment(value).add(-1, 'month').format('YYYY-MM'));
  };

  const handleNextClick = () => {
    onChange(moment(value).add(1, 'month').format('YYYY-MM'));
  };

  return (
    <div className={classNames(styles.root, className)}>
      <Input
        className={styles.input}
        onChange={handleChange}
        type="month"
        {...inputProps}
      />

      <div className={styles.value}>
        <span className={styles.valueMonth}>
          {moment(value).format('MMMM')}
        </span>
        <span className={styles.valueYear}>{moment(value).format('YYYY')}</span>
      </div>

      <button
        aria-label="Prev month"
        className={classNames(styles.arrow, styles.arrowPrev)}
        disabled={moment(value).isSameOrBefore(min)}
        onClick={handlePrevClick}
        type="button"
      >
        <ArrowComponent className={styles.arrowIcon} />
      </button>

      <button
        aria-label="Next month"
        className={classNames(styles.arrow, styles.arrowNext)}
        onClick={handleNextClick}
        type="button"
      >
        <ArrowComponent className={styles.arrowIcon} />
      </button>
    </div>
  );
}

export default AmountInput;
