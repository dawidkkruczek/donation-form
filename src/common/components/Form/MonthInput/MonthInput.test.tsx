import React from 'react';
import { render } from '@testing-library/react';
import MonthInput from './MonthInput';

describe('common/components/MonthInput', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<MonthInput />);
    expect(asFragment()).toMatchSnapshot();
  });
});
