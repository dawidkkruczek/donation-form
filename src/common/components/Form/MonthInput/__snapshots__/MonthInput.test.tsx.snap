// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/components/MonthInput should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <input
      class="root input"
      type="month"
    />
    <div
      class="value"
    >
      <span
        class="valueMonth"
      >
        February
      </span>
      <span
        class="valueYear"
      >
        2024
      </span>
    </div>
    <button
      aria-label="Prev month"
      class="arrow arrowPrev"
      disabled=""
      type="button"
    >
      <svg
        class="arrowIcon"
      >
        arrow.svg
      </svg>
    </button>
    <button
      aria-label="Next month"
      class="arrow arrowNext"
      type="button"
    >
      <svg
        class="arrowIcon"
      >
        arrow.svg
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
